### PR TITLE
ROX-19696: add functions to parse deduper state

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

This PR adds functions to parse the deduper state. Follow-up PRs will be opened to call this functions and initialize the deduper state.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

* [x] Unit tests added
* [x] CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: sensor/common/deduper/key.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		&central.SensorEvent_ComplianceOperatorResult{},
+		&central.SensorEvent_ComplianceOperatorProfile{},
+		&central.SensorEvent_ComplianceOperatorRule{},
+		&central.SensorEvent_ComplianceOperatorScanSettingBinding{},
+		&central.SensorEvent_ComplianceOperatorScan{},
+		&central.SensorEvent_AlertResults{},
+	}
+)
+
+// Key is the Key by which messages are deduped.
```
Comment (fredrb) [inline sensor/common/deduper/key.go:38]:
```suggestion
// Key by which messages are deduped.
```

File: sensor/common/deduper/key.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+type Key struct {
+	ID           string
+	ResourceType reflect.Type
+}
+
+// CopyDeduperState makes a copy of the deduper state.
+func CopyDeduperState(state map[string]uint64) map[Key]uint64 {
```
Comment (fredrb) [inline sensor/common/deduper/key.go:45]: Maybe `ParseDeduperState` or `ConvertDeduperState` would make more sense? I think if you're converting from type `A` to type `B` it's kinda implicit that the result is a copy?

File: sensor/common/deduper/key.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	if state == nil {
+		return make(map[Key]uint64)
+	}
+
+	result := make(map[Key]uint64, len(state))
+	for k, v := range state {
+		parsedKey, err := keyFrom(k)
+		if err != nil {
+			log.Warnf("Deduper state received from central has malformed entry: %s->%d: %s", k, v, err)
```
Comment (fredrb) [inline sensor/common/deduper/key.go:54]: nit: if we think of this code as a library, maybe remove the assumption that the state came from central? 
```suggestion
			log.Warnf("Deduper state has malformed entry: %s->%d: %s", k, v, err)
```

File: sensor/common/deduper/key_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	"testing"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/fixtures/fixtureconsts"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	randomID   = "2188759360372708523"
+	randomHash = 2188759360372708523
```
Comment (fredrb) [inline sensor/common/deduper/key_test.go:15]: These don't seem random :-) 

How about `StubID` and `StubHash`?

File: sensor/common/deduper/key_test.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+type deduperKeySuite struct {
+	suite.Suite
+}
+
+func Test_DeduperKeySuite(t *testing.T) {
+	suite.Run(t, new(deduperKeySuite))
+
+}
+
+func (s *deduperKeySuite) Test_CopyState() {
```
Comment (fredrb) [inline sensor/common/deduper/key_test.go:74]: Great job on these tests!

**Review by fredrb — APPROVED** [review]
Looks great to me. Only a couple of nits.

File: sensor/common/deduper/key.go (hunk 1 above)
Comment (lvalerom) [reply sensor/common/deduper/key.go]: Done

File: sensor/common/deduper/key.go (hunk 2 above)
Comment (lvalerom) [reply sensor/common/deduper/key.go]: Done

File: sensor/common/deduper/key.go (hunk 3 above)
Comment (lvalerom) [reply sensor/common/deduper/key.go]: Done

File: sensor/common/deduper/key_test.go (hunk 4 above)
Comment (lvalerom) [reply sensor/common/deduper/key_test.go]: Done

File: sensor/common/deduper/key_test.go (hunk 5 above)
Comment (lvalerom) [reply sensor/common/deduper/key_test.go:74]: 🙂

**Pull request merged**